### PR TITLE
Unit test expr_skeleton and move it to its own file

### PR DESCRIPTION
**Pull request by romainbrenguier**
Move the class to its own files and unit test it.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 9c31050261b5764ec4a8820e301b8c7b0fada0d5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117126740
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Comment by romainbrenguier** [comment]
@smowton I've updated the method names

**Review by smowton — APPROVED** [review]

File: unit/goto-symex/expr_skeleton.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#include <goto-symex/expr_skeleton.h>
+#include <util/arith_tools.h>
+#include <util/c_types.h>
+#include <util/std_expr.h>
+
+SCENARIO("expr skeleton", "[core][goto-symex][symex-assign][expr-skeleton]")
+{
+  const symbol_exprt foo{"foo", typet{}};
+  GIVEN("Skeletons `☐`, `☐[index]` and `☐.field1`")
```
Comment (majakusber) [inline unit/goto-symex/expr_skeleton.cpp:20]: :pick: Why use the symbol here? Seems strange..

**Review by majakusber — APPROVED** [review]

File: unit/goto-symex/expr_skeleton.cpp (hunk 1 above)
Comment (romainbrenguier) [reply unit/goto-symex/expr_skeleton.cpp:20]: Do you mean ☐? It is used in the same way as in the documentation https://github.com/diffblue/cbmc/pull/4843/files#diff-003f21a13603cd3a681d0d0f440e114bR20

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 948e9f46e3f4292f7e01ef0c16b98d734b7d89cd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117637722

**Pull request merged**